app/frontend/src/features/profile/view/leaveReference/formSteps/submit/SubmitReference.tsx

```diff
@@ -0,0 +1,186 @@
+import { useMediaQuery, useTheme } from "@material-ui/core";
+import Alert from "components/Alert";
+import Button from "components/Button";
+import {
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+} from "components/Dialog";
+import {
+  HOST_REQUEST_REFERENCE_EXPLANATION,
+  HOST_REQUEST_REFERENCE_SUCCESS_DIALOG,
+  OKAY,
+  REFERENCE_SUCCESS,
+  SUBMIT,
+} from "features/profile/constants";
+import {
+  useWriteFriendReference,
+  useWriteHostReference,
+} from "features/profile/hooks/referencesHooks";
+import ReferenceOverview from "features/profile/view/leaveReference/formSteps/submit/ReferenceOverview";
+import {
+  ReferenceContextFormData,
+  useReferenceStyles,
+} from "features/profile/view/leaveReference/ReferenceForm";
+import { User } from "pb/api_pb";
+import { ReferenceType } from "pb/references_pb";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { useHistory, useParams } from "react-router";
+import { baseRoute, referenceTypeRoute, userRoute } from "routes";
+import {
+  WriteFriendReferenceInput,
+  WriteHostRequestReferenceInput,
+} from "service/references";
+
+export interface SubmitReferenceProps {
+  user: User.AsObject;
+  referenceData: ReferenceContextFormData;
+}
+
+export default function SubmitReference({
+  user,
+  referenceData,
+}: SubmitReferenceProps) {
+  const {
+    writeFriendReference,
+    reset: resetFriendReferenceWriting,
+    error: friendReferenceError,
+    isLoading: isFriendReferenceLoading,
+  } = useWriteFriendReference(user.userId);
+  const {
+    writeHostRequestReference,
+    reset: resetHostRequestReferenceWriting,
+    error: hostRequestReferenceError,
+    isLoading: isHostRequestReferenceLoading,
+  } = useWriteHostReference(user.userId);
+
+  const { referenceType, hostRequestId } =
+    useParams<{
+      referenceType: string;
+      hostRequestId?: string;
+    }>();
+  const theme = useTheme();
+  const classes = useReferenceStyles();
+  const history = useHistory();
+  const [isOpen, setIsOpen] = useState(false);
+  const isSmOrWider = useMediaQuery(theme.breakpoints.up("sm"));
+  const { handleSubmit } = useForm<ReferenceContextFormData>();
+
+  const onFriendReferenceSubmit = () => {
+    const formData: WriteFriendReferenceInput =
+      referenceData.wasAppropriate === "true"
+        ? {
+            toUserId: user.userId,
+            wasAppropriate: true,
+            text: referenceData.text,
+            rating: referenceData.rating,
+          }
+        : {
+            toUserId: user.userId,
+            wasAppropriate: false,
+            text: referenceData.text,
+            rating: referenceData.rating,
+          };
+
+    resetFriendReferenceWriting();
+    writeFriendReference(
+      {
+        referenceData: formData,
+      },
+      {
+        onSuccess: () => {
+          history.push(`${userRoute}/${user.username}/references`);
```

Comment:
```suggestion
          history.push(`${userBaseRoute}/${user.username}/references`);
```
This was taking me to something like `/user/:username?/:tab?/darren/references` otherwise

Reply:
Is there a good reason why this differs from the host reference flow which opens a confirmation dialog?

Reply:
Yes. First it was just a success alert for both but we decided to change it to this. The friend references are immediately available and therefore we just redirect to their reference page where it is now visible. The pop up dialog was made to let users know host request references are available when both users have done it or when 2 weeks have passed.